source/fundamentals/crud/read-operations/text.txt

```diff
@@ -0,0 +1,237 @@
+   { "_id": 4, "course": "Computer Science" }
+   { "_id": 5, "course": "Computer Hacking Revealed" }
+   { "_id": 6, "course": "Advanced Computer Game Programming" }
+   { "_id": 7, "course": "Image Processing in Computer Science" }
+
+.. note::
+
+    The output of the snippet above is equivalent to performing the
+    following text search:
+
+    .. literalinclude:: /includes/fundamentals/code-snippets/SearchText.java
+       :language: java
+       :dedent:
+       :start-after: begin multiWordExample
+       :end-before: end multiWordExample
+
+By a Phrase
+~~~~~~~~~~~
+
+To search for a phrase in your text index, pass the phrase with escaped quotes
+to the ``text()`` method.  
+
+If you don't add escaped quotes, the method performs a logical OR on the
+terms listed.
+
+Example 
+```````
+
+The student would like to know which courses mention "Computer Science"
+in their name. 
+
+To address this scenario, the student performs a text search on
+documents that contain the phrase "Computer Science":
+
+.. literalinclude:: /includes/fundamentals/code-snippets/SearchText.java
+   :language: java
+   :dedent:
+   :start-after: begin phraseExample
+   :end-before: end phraseExample
+
+The following shows the output of the code above:
+
+.. code-block:: json
+   :copyable: false
+
+   { "_id": 2, "course": "Discrete Mathematical Structures for Computer Science" }
+   { "_id": 4, "course": "Computer Science" }
+   { "_id": 7, "course": "Image Processing in Computer Science" }
+
+Excluding Terms
+~~~~~~~~~~~~~~~
+
+To exclude a term in your text search, prefix a minus sign, ``-``, to
+each term you want to omit. 
+
+Example 
+```````
+The student would like to know which courses mention "Computer" without
+"Science".
+
+To address this scenario, the student performs a text search on
+documents that contain the term "Computer", but not contain the term
+"Science": 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/SearchText.java
+   :language: java
+   :dedent:
+   :start-after: begin negateExample
+   :end-before: end negateExample
+
+The following shows the output of the code above:
+
+.. code-block:: json
+   :copyable: false
+
+   { "_id": 5, "course": "Computer Hacking Revealed" }
+   { "_id": 6, "course": "Advanced Computer Game Programming" }
+
+Sort by Relevance
+~~~~~~~~~~~~~~~~~
+
+You can see how relevant the documents returned by the text are by
+including a projection to calculate the ``metaTextScore``. The
+``metaTextScore`` signifies how well the document matched the search
+term.
+
+Example 
+```````
+The student would like to see which course mentioning "Computer Science"
+is the most relevant. 
+
+To address this scenario, the student performs the following actions:
+
+- A text search for the term "Computer Science"
+- A projection to calculate the ``metaTextScore`` and place the score in a field called ``score``
+- A sort to display the results in descending order
```

Comment:
Remember that we shouldn't assume our readers know about any of the methods or classes because they could have arrived at the documentation from anywhere. We should reference Projections and Sorts builders somewhere, ideally before we talk about projections and sorts. We should also add API documentation links at the end of this section.